### ISSUE TITLE
Use coding style standardjs

### ISSUE THREAD

**Issue by Dexus** [haraka/Haraka#2113]
use of https://standardjs.com coding style

It that not full match the needs, we can extend the rules of the eslint package of standardjs
What do you thing about it?

**Comment by KingNoosh** [comment]
I'd rather prefer airbnb style but both of those styles are a big drastic change.

We should agree on a final ruleset maybe, I'd rather be a similar style to the rest of the open source nodejs projects then completely different though. Which is why I would agree on either Standard/AirBnB ruleset.

**Comment by smfreegard** [comment]
I think all of this is just a massive distraction.

We should be improving the software (read: making it faster, adding new features, fixing shortcomings) rather than concentrating on changing the style of the code.

Implementing this would create a massive patch-bomb (e.g. removing semi-colons).

Personally speaking - Haraka is already missing a bunch of code that I wrote years ago because of changes like this making it so time consuming to keep things in-sync that I stopped trying.  I'll get around it it eventually but spare time is precious these days.

**Comment by KingNoosh** [comment]
I agree with the massive patch-bomb, maybe something as drastic as changing the style should be done for v3 instead?

**Comment by smfreegard** [comment]
> I agree with the massive patch-bomb, maybe something as drastic as changing the style should be done for v3 instead?

Yeah - I'd have no problem if we did it that way.   It would have to be done just before v3 is released though as trying to keep the v3 repo in step would become impossible otherwise.

**Comment by baudehlo** [comment]
We are not changing the coding style.

**Comment by KingNoosh** [comment]
So we're sticking with using callbacks for v3? Not promises or async/await?

**Comment by msimerson** [comment]
We've talked about changing the code style in the past (I favored it) and I've come around to mostly agree with @smfreegard and @baudehlo that we should **not** change the style. While Haraka's current "style" is not very stylish, and it's certainly not *my* preferred style, it is the style we have. Changing it comes with costs (as @smfreegard raised) and changing it should only be done for very clear and compelling reasons.

> So we're sticking with using callbacks for v3? Not promises or async/await?

Again, a dose of pragmatism is in order. We've got callbacks now but it's **finally** clear that promises are the future, and async/await is built on promises (there's a pun in there). Callbacks work, they're in place, and they're very fast. At the moment, they're still faster than promises, so that's one potential drawback to using them. However, promises are much easier to reason about. Therefore, it seems sensible that we begin using promises/async/await in code paths that have little impact on performance.

By the time Haraka has evolved enough to use promises everywhere, we'll be on node 10 or 12, all the libraries we depend on will have evolved to support promises, and the performance issues will likely have faded.
